account_invoice_allowed_product/models/res_partner.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    supplier_invoice_only_allowed = fields.Boolean(
```

Comment:
only_allowed_products_supplier_invoice ?
(for a consistent naming with `only_allowed_products = fields.Boolean()` )